core/src/main/java/nu/yona/server/analysis/entities/ActivityCommentMessage.java

```diff
@@ -159,8 +160,9 @@ public void prepareForDelete()
 		{
 			// This is the sender copy, so do the work
 			Set<Message> messagesToBeCascadinglyDeleted = getMessagesToBeCascadinglyDeletedForJustThisObject();
-			messagesToBeCascadinglyDeleted.addAll(buddyMessage.getMessagesToBeCascadinglyDeletedForJustThisObject());
-			messagesToBeCascadinglyDeleted.add(buddyMessage);
+			ActivityCommentMessage loadedBuddyMessage = EntityUtil.enforceLoading(buddyMessage);
```

Comment:
Is this a bugfix?

Reply:
I accidentally created this change set based on another branch, which has this fix. Let's merge the other PRs first and then merge back into this branch. That should make this change go away.

Reply:
Sorry, no. This is indeed a bug fix. This I mentioned in my first commit and should have mentioned in the PR text:
> * We have an old way of explicitly loading an entity. Nowadays, Hibernate.unproxy exists, so we should adopt that
> * In ActivityCommentMessage.getMessagesToBeCascadinglyDeleted, we were using a non-loaded Hibernate proxy. This was masked because the default constructor (required for JPA) of BuddyMessage used the wrong super constructor, thus causing the replies collection to be initialized. This is corrected now.

Reply:
Updated the PR description.